rcl/src/rcl/lexer_lookahead.c

```diff
@@ -222,8 +222,14 @@ rcl_lexer_lookahead2_expect(
     return ret;
   }
   if (type != lexeme) {
+    if (RCL_LEXEME_NONE == lexeme || RCL_LEXEME_EOF == lexeme) {
+      RCL_SET_ERROR_MSG_WITH_FORMAT_STRING(
+        buffer->impl->allocator, "Expected lexeme type (%d) not found", type);
+      return RCL_RET_WRONG_LEXEME;
+    }
     RCL_SET_ERROR_MSG_WITH_FORMAT_STRING(
-      buffer->impl->allocator, "Expected %d got %d at %lu", type, lexeme, buffer->impl->text_idx);
+      buffer->impl->allocator, "Expected lexeme type %d, got %d at index %lu", type, lexeme,
```

Comment:
Certainly just printing the type as a number instead of a string was the main motivation for separating the messages (because I had to go find the type to come to the conclusion that it wasn't found), but even printing strings I think it'd be clearer to the user to say explicitly that it wasn't found (as opposed to them having to infer it from the types). I also thought that it might deserve a different return code in some situations but that's just hypothetical :)